packages/Textarea/src/textarea.scss

```diff
@@ -2,94 +2,145 @@
+
+    /* Placeholder */
+    --ids-textarea-color-placeholder: var(--hop-neutral-text-weakest);
+
+    /* Focus */
+    --ids-textarea-focus-border-color: var(--hop-neutral-border-active);
+    --ids-textarea-focus-box-shadow: 0 0 0 0.12rem var(--hop-neutral-surface), 0 0 0 0.219rem var(--hop-primary-border-focus);
+
+    /* Hover */
+    --ids-textarea-hover-border-color: var(--hop-neutral-border-hover);
+    --ids-textarea-hover-background-color: var(--hop-neutral-surface-hover);
+
+    /* Error */
+    --ids-textarea-error-border-color: var(--hop-danger-border-strong);
+
+    /* Disabled */
+    --ids-textarea-disabled-background: var(--hop-neutral-border-disabled);
+    --ids-textarea-disabled-color: var(--hop-neutral-surface-disabled);
+
+    /* Character Indicator */
+    --ids-textarea-char-indicator-bottom: 0.5rem;
+    --ids-textarea-char-indicator-right: 1rem;
+    --ids-textarea-char-indicator-color: var(--hop-neutral-text-weakest);
+    --ids-textarea-char-indicator-size: var(--hop-body-xs-font-size);
+    --ids-textarea-char-indicator-line-height: var(--hop-body-xs-line-height);
 }
 
 .ids-textarea {
-  font-family: var(--ids-textarea-font-family);
-  font-size: var(--ids-textarea-font-size);
-  position: relative;
-
-  &,
-  & * {
-    box-sizing: border-box;
-  }
-
-  &__field {
-    border: .1rem solid var(--ids-textarea-border-color);
-    border-radius: var(--ids-textarea-border-radius);
-    color: var(--ids-textarea-color);
-    display: block;
-    font-family: inherit;
-    font-size: inherit;
-    line-height: var(--ids-textarea-line-height);
-    outline: none;
-    padding: var(--ids-textarea-padding);
-    resize: none;
-    width: 100%;
-
-    @include mixins.fancy-webkit-scrollbar;
-
-    &::placeholder {
-      color: var(--ids-textarea-color-placeholder);
-    }
+    font-family: var(--ids-textarea-font-family);
+    font-size: var(--ids-textarea-font-size);
+    position: relative;
 
-    &:focus,
-    .ids-textarea--focus & {
-      border-color: var(--ids-textarea-focus-border-color);
+    &,
+    & * {
+        box-sizing: border-box;
     }
 
-    &--error {
-      border-color: var(--ids-textarea-error-border-color);
+    &__field {
+        border: .1rem solid var(--ids-textarea-border-color);
+        border-radius: var(--ids-textarea-border-radius);
+        color: var(--ids-textarea-color);
+        display: block;
+        font-family: inherit;
+        font-size: inherit;
+        line-height: var(--ids-textarea-line-height);
+        outline: none;
+        padding: var(--ids-textarea-padding);
+        resize: none;
+        width: 100%;
+
+        @include mixins.fancy-webkit-scrollbar;
+
+        &::placeholder {
+            color: var(--ids-textarea-color-placeholder);
+        }
+
+        &:focus:not(&--error),
+        .ids-textarea--focus:not(&--error) & {
+            border-color: var(--ids-textarea-focus-border-color);
+        }
+
+        &:hover:not(:disabled) {
```

Comment:
I prefer not to use :not if it's not needed and can just be overridden normally. The reason being that it becomes more specific and can end up being more annoying later down the road when others use it.

Reply:
I get where you are coming from. Not is my best friend especially for disabled state, I'll override the disabled hover, I tend to prefer to avoid adding unnecessary CSS but I get the why.